.github/workflows/update-public-instances.yml

```
@@ -0,0 +1,48 @@
+name: Update public instances
+on:
+  push:
+    branches:
+      - instance-picker
```

Because of this, your CI run doesn't count as a PR run. You need to either change the `github.event_name` check below to `push`, or change this to

```suggestion
  pull_request:
    branches:
      - main
```

(triggered by  PR to the main branch)